operator/pkg/helmreconciler/reconciler.go

```diff
@@ -247,9 +247,9 @@ func (h *HelmReconciler) processRecursive(manifests name.ManifestMap) *v1alpha1.
 // CheckSSAEnabled is a helper function to check whether ServerSideApply should be used when applying manifests.
 func (h *HelmReconciler) CheckSSAEnabled() bool {
 	if h.kubeClient != nil {
-		// There is a mutatingwebhook in gke that would corrupt the managedFields, which is fixed in k8s 1.18.
-		// See: https://github.com/kubernetes/kubernetes/issues/96351
-		if kube.IsAtLeastVersion(h.kubeClient, 18) {
+		// SSA went GA in k8s 1.22. Earlier versions (eg: 1.18) have an issue where managedFields is not present on upgraded clusters.
+		// See: https://github.com/istio/istio/issues/37946#issuecomment-1072875625
+		if kube.IsAtLeastVersion(h.kubeClient, 22) {
```

Comment:
can we do `if >22 { return true } else if > 18 { detect } else { return false }`

Right now users on 1.18-1.21 may be broken by this change

Reply:
Do you think it is worth adding a reconciler option to disable SSA ? 🤔 
Can't see why someone would want to do it but checking anyways.